### PR TITLE
Add queries test

### LINKED ISSUE #84: Add tests for queries
The tests for the queries were removed on the merge of [pr 70](https://github.com/runtimerevolution/revent-api/pull/70) due to the changes made primarily to the Picture model and implementation of Localstack.

To see the structure of a test refer to the [strawberry documentation](https://strawberry.rocks/docs/operations/testing).

## Requirements
- Each query must be tested to ensure everything is working properly

### PR DESCRIPTION
**Pull request by tiagocardoso98**
Closes #84 
- Add tests for all queries
- Fix factory warnings of deprecation

### CODE REVIEW

File: photo/types.py [hunk 1]
```diff
@@ -15,6 +16,7 @@
 
 @strawberry.django.type(User)
 class UserType:
+    id: uuid.UUID
```
Comment (claudio-vagarinho) [inline photo/types.py:19]: this will conflict with the changes I made in the UserType. @kenvontucky should this be a string or uuid?

File: photo/tests/factories.py [hunk 2]
```diff
@@ -17,6 +17,7 @@
 class UserFactory(factory.django.DjangoModelFactory):
     class Meta:
         model = User
+        skip_postgeneration_save = True
```
Comment (claudio-vagarinho) [inline photo/tests/factories.py:20]: @JdFSilva questioned why this was added

**Review by claudio-vagarinho — COMMENTED** [review]
Regarding the changes to the queries that now use the user id instead of the email, they will also need to be reflected on the front end correct? do we want this to be the case?

File: photo/tests/factories.py (hunk 2 above)
Comment (tiagocardoso98) [reply photo/tests/factories.py:20]: This was to resolve a deprecation warning that was being raised by all factories using post_generation.

File: photo/types.py (hunk 1 above)
Comment (tiagocardoso98) [reply photo/types.py:19]: uuid.UUID will behave as string when queried.

File: photo/queries.py [hunk 3]
```diff
@@ -24,7 +25,9 @@
 @strawberry.type
 class Query:
     @strawberry.field
-    def users(self, email: str = None) -> List[UserType]:
+    def users(self, user: uuid.UUID = None, email: str = None) -> List[UserType]:
+        if user:
```
Comment (kenvontucky) [inline photo/queries.py:29]: This should only use the user id, and must be restricted to the user, meaning that a user cannot see the details of another person

File: photo/queries.py [hunk 4]
```diff
@@ -37,49 +40,49 @@ def pictures(self, picture: int = None) -> List[PictureType]:
...
-        self, id: int = None, user_email: str = None, picture_id: int = None
+        self, id: int = None, user: uuid.UUID = None, picture_id: int = None
     ) -> List[PictureCommentType]:
         if id:
             return PictureComment.objects.filter(id=id)
-        if user_email and picture_id:
-            return PictureComment.objects.filter(user=user_email, picture=picture_id)
-        elif user_email:
-            return PictureComment.objects.filter(user=user_email)
+        if user and picture_id:
```
Comment (kenvontucky) [inline photo/queries.py:47]: Lets refactor this to required the picture id only

File: photo/queries.py [hunk 5]
```diff
@@ -37,49 +40,49 @@ def pictures(self, picture: int = None) -> List[PictureType]:
...
+        elif user:
+            return PictureComment.objects.filter(user=user)
         elif picture_id:
             return PictureComment.objects.filter(picture=picture_id)
         return PictureComment.objects.all()
 
     @strawberry.field
     def collections(
-        self, user_email: str = None, name: str = None, id: int = None
+        self, user: uuid.UUID = None, name: str = None, id: int = None
```
Comment (kenvontucky) [inline photo/queries.py:57]: this should only return `return Collection.objects.all()` all the remaining must be done with graphql filters

File: photo/queries.py [hunk 6]
```diff
@@ -37,49 +40,49 @@ def pictures(self, picture: int = None) -> List[PictureType]:
...
 
     @strawberry.field
-    def contests(self, user_email: str = None, id: int = None) -> List[ContestType]:
+    def contests(self, user: uuid.UUID = None, id: int = None) -> List[ContestType]:
         if id:
             return Contest.objects.filter(id=id)
-        if user_email:
-            return Contest.objects.filter(created_by=user_email)
+        if user:
+            return Contest.objects.filter(created_by=user)
```
Comment (kenvontucky) [inline photo/queries.py:74]: this should only do `return Contest.objects.all()`

**Review by kenvontucky — CHANGES_REQUESTED** [review]

File: photo/tests/factories.py (hunk 2 above)
Comment (tiagocardoso98) [reply photo/tests/factories.py:20]: See the deprecation section of https://github.com/FactoryBoy/factory_boy/blob/master/docs/changelog.rst#330-unreleased

File: photo/queries.py (hunk 3 above)
Comment (tiagocardoso98) [reply photo/queries.py]: Should this restriction to the user only being able to see his detais be implemented when implementing authentication?

File: photo/queries.py (hunk 5 above)
Comment (tiagocardoso98) [reply photo/queries.py:57]: Implemented

File: photo/queries.py (hunk 6 above)
Comment (tiagocardoso98) [reply photo/queries.py:74]: Implemented

File: photo/queries.py [hunk 7]
```diff
@@ -37,49 +40,49 @@ def pictures(self, picture: int = None) -> List[PictureType]:
...
         if id:
             return ContestSubmission.objects.filter(id=id)
-        if user_email or contest:
-            if user_email:
-                return ContestSubmission.objects.filter(picture__user__email=user_email)
+        if user or contest:
+            if user:
+                return ContestSubmission.objects.filter(picture__user=user)
             return ContestSubmission.objects.filter(contest__id=contest)
         return ContestSubmission.objects.all()
```
[the remark this answers is not in the document]
Comment (tiagocardoso98) [reply photo/queries.py:87]: Implemented

File: photo/queries.py (hunk 4 above)
Comment (tiagocardoso98) [reply photo/queries.py]: Implemented

**Review by kenvontucky — CHANGES_REQUESTED** [review]
Lets incorporate ContestSearch within the Contest, there's no point on having two things graphql types on the same thing.

**Review by claudio-vagarinho — APPROVED** [review]

**Review by claudio-vagarinho — APPROVED** [review]

File: photo/tests/test_queries/test_collection.py [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+from photo.tests.factories import CollectionFactory, PictureFactory, UserFactory
+from photo.tests.test_queries.graphql_queries import (
+    collections_query_all,
+    collections_query_filter,
+)
+
+
+class CollectionTest(TestCase):
+    def setUp(self):
+        self.batch = 10
```
Comment (kenvontucky) [inline photo/tests/test_queries/test_collection.py:14]: this can be a class variable called batch_size

File: photo/tests/test_queries/test_collection.py [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+        user = UserFactory(user_profile_picture=True)
+        self.pictures = PictureFactory.create_batch(self.batch, user=user)
+        self.collections = CollectionFactory.create_batch(
+            self.batch, collection_pictures=self.pictures
+        )
+
+    def test_query_all(self):
```
Comment (kenvontucky) [inline photo/tests/test_queries/test_collection.py:21]: since a query returns everything this test should be called test_query_success or test_query_with_data, and there should another test for an empty query, test_query_without_data

File: photo/tests/test_queries/test_collection.py [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+                [
+                    field.name
+                    for field in (
+                        Collection._meta.fields + Collection._meta.many_to_many
+                    )
+                ]
+            ),
+        )
+
+    def test_query_one(self):
```
Comment (kenvontucky) [inline photo/tests/test_queries/test_collection.py:42]: here we're testing a filter so the name should be test_filter_by_<the filter being tested>

File: photo/tests/test_queries/test_collection.py [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+
+        self.assertEqual(result.errors, None)
+        self.assertEqual(len(result.data["collections"]), 1)
+        self.assertEqual(result.data["collections"][0]["name"], collection.name)
+        self.assertEqual(
+            result.data["collections"][0]["user"]["id"], str(collection.user.id)
+        )
+        self.assertEqual(len(result.data["collections"][0]["pictures"]), len(pictures))
+
+    def test_query_user_name(self):
```
Comment (kenvontucky) [inline photo/tests/test_queries/test_collection.py:62]: same here

File: photo/tests/test_queries/test_collection.py [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+
+        self.assertEqual(result.errors, None)
+        self.assertEqual(len(result.data["collections"]), 1)
+        self.assertEqual(result.data["collections"][0]["name"], collection.name)
+        self.assertEqual(
+            result.data["collections"][0]["user"]["id"], str(collection.user.id)
+        )
+        self.assertEqual(len(result.data["collections"][0]["pictures"]), len(pictures))
+
+    def test_query_by_name(self):
```
Comment (kenvontucky) [inline photo/tests/test_queries/test_collection.py:85]: same here

File: photo/tests/test_queries/test_collection.py [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+
+        self.assertEqual(result.errors, None)
+        self.assertEqual(len(result.data["collections"]), 1)
+        self.assertEqual(result.data["collections"][0]["name"], collection.name)
+        self.assertEqual(
+            result.data["collections"][0]["user"]["id"], str(collection.user.id)
+        )
+        self.assertEqual(len(result.data["collections"][0]["pictures"]), len(pictures))
+
+    def test_query_by_user(self):
```
Comment (kenvontucky) [inline photo/tests/test_queries/test_collection.py:103]: same here

**Review by kenvontucky — CHANGES_REQUESTED** [review]
update the remaining test following the remarks left in the test_collection.py

File: photo/tests/test_queries/test_collection.py (hunk 9 above)
Comment (tiagocardoso98) [reply photo/tests/test_queries/test_collection.py]: Applied change
Added tests for test_query_without_data

File: photo/tests/test_queries/test_collection.py (hunk 11 above)
Comment (tiagocardoso98) [reply photo/tests/test_queries/test_collection.py]: Applied change

File: photo/tests/test_queries/test_collection.py (hunk 12 above)
Comment (tiagocardoso98) [reply photo/tests/test_queries/test_collection.py]: Applied change

File: photo/tests/test_queries/test_collection.py (hunk 13 above)
Comment (tiagocardoso98) [reply photo/tests/test_queries/test_collection.py]: Applied change

**Review by kenvontucky — APPROVED** [review]

File: photo/tests/test_database/test_contestSubmission.py
Comment (hugo-serra) [inline photo/tests/test_database/test_contestSubmission.py:1]: the file name should be either camelCase or snake_case not both

File: photo/tests/test_queries/test_contestSubmission.py
Comment (hugo-serra) [inline photo/tests/test_queries/test_contestSubmission.py:1]: filename should be snake_case or camelCase

Comment (hugo-serra) [reply photo/tests/test_queries/test_contestSubmission.py:1]: since we're using snake_case everywhere else, we should continue to use snake_case
